### PR TITLE
Transport acks != application processing

### LINKED ISSUE #15: Exposing datagram acknowledgements
An issue came up in discussions around WebTransport: https://github.com/w3c/webtransport/issues/168

The problem is that datagram frames may be acknowledged for the sake of congestion control, but dropped for the sake of flow control. 

A receiver with a full datagram buffer would have two choices:
1. Drop any packets with datagram frames and do not acknowledge them.
2. Drop the datagram frames but acknowledge the packets.

The first approach seems terrible because it drop any other frames bundled with the datagram (ex. STREAM) and cause retransmissions. It would also be treated as network congestion by the sender, causing limited bandwidth for otherwise unrelated STREAM frames.

The second approach has ramifications for any implementations that MAY expose datagram acknowledgements. Specifically, the sender would believe that the datagram has been received (although not necessary processed), when in reality it could have been dropped and will never be processed.

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #15

### CODE REVIEW

**Review by nibanks — APPROVED** [review]
Text is good, but I am still a bit sad that we don't have some way to leverage the QUIC ACKs in a general sense (not implementation specific).

File: draft-ietf-quic-datagram.md [hunk 1]
```diff
@@ -249,11 +249,17 @@ acknowledgements is not used for loss recovery.
...
 have been lost, the implementation MAY notify the application that it believes
-the datagram was lost. Similarly, if a packet containing a DATAGRAM frame is
-acknowledged, the implementation MAY notify the application that the datagram
-was successfully transmitted and received. Note that, due to reordering, a
-DATAGRAM frame that was thought to be lost could at a later point be received
-and acknowledged.
+the datagram was lost.
+
+Similarly, if a packet containing a DATAGRAM frame is acknowledged, the
+implementation MAY notify the application that the datagram was successfully
```
Comment (LPardue) [inline draft-ietf-quic-datagram.md]: which application? The sender, receiver, or both?

Comment (LPardue) [reply draft-ietf-quic-datagram.md]: I appreciate this text was already like this before the PR. However, IMO it wouldn't hurt to be specific here, the following sentences are. 

This isn't a blocking issue, so go ahead and merge if you disagree.

Comment (DavidSchinazi) [reply draft-ietf-quic-datagram.md]: I agree with Lucas, how about:
```suggestion
implementation MAY notify the sender application that the datagram was successfully
```
same for line 251 above.

**Review by DavidSchinazi — APPROVED** [review]

**Pull request merged**